pkg/remotewrite/trend.go

```diff
@@ -1,84 +1,66 @@
 package remotewrite
 
 import (
-	"math"
-	"sort"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
 	"go.k6.io/k6/metrics"
 )
 
-// The following functions are an attempt to add ad-hoc optimization to TrendSink,
-// and are a partial copy-paste from k6/metrics.
-// TODO: re-write & refactor this once metrics refactoring progresses in k6.
-
 type trendSink struct {
```

Comment:
What do you think if we split implementations? 

Like current terndSink can be an old version and introduce another `trendNativeHistogram.` 

That way, we can avoid this all `if t.H == nil { ... return }`, which makes code more straightforward.